x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -179,6 +208,56 @@ export interface FindOptions extends IndexType {
   filter?: string;
 }
 
+type BulkEditActionRuleFields = keyof Pick<
+  Rule,
+  'actions' | 'schedule' | 'tags' | 'throttle' | 'notifyWhen'
+>;
+
+export type BulkEditActionRule =
+  | {
+      action: 'add' | 'delete' | 'set';
+      field: Extract<BulkEditActionRuleFields, 'tags'>;
+      value: string[];
+    }
+  | {
+      action: 'add' | 'set';
+      field: Extract<BulkEditActionRuleFields, 'actions'>;
+      value: NormalizedAlertAction[];
+    };
+
+// schedule, throttle, notifyWhen is commented out before https://github.com/elastic/kibana/issues/124850 will be implemented
+// | {
+//     action: 'set';
+//     field: Extract<BulkEditActionRuleFields, 'schedule'>;
+//     value: Rule['schedule'];
+//   }
+// | {
+//     action: 'set';
+//     field: Extract<BulkEditActionRuleFields, 'throttle'>;
+//     value: Rule['throttle'];
+//   }
+// | {
+//     action: 'set';
+//     field: Extract<BulkEditActionRuleFields, 'notifyWhen'>;
+//     value: Rule['notifyWhen'];
+//   };
+
+type RuleParamsModifier<Params extends RuleTypeParams> = (params: Params) => Params;
+
+export interface BulkEditOptions<Params extends RuleTypeParams> {
+  filter: string | KueryNode;
+  actions: BulkEditActionRule[];
```

Comment:
nit -> it might be a good idea to call it `editActions` to avoid confusing with all of our actions terminology

Reply:
done